docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -32,197 +35,203 @@ note over ERCC1, ERCC2 #FFaa00
      be dependent on such policy.
 end note
 
-group enclave creation
-  activate Admin
-
-  note over Admin #FFaa00
-    These operations are assumed to be performed only after the chaincode definition is
-    committed on the channel for two reasons:
-    1) chaincode invocation may not be available before the definition is committed,
-    2) enclave registration should be feasible only for committed chaincodes
-
-    PS: in a future tighter integration with fabric we could remove the first constraint
-    and could potentially piggyback the steps before the ercc registration (enclave and
-    evidence generation) in the `approveformyorg` and the registration as part of
-    `commit` without having to change the protocol flows.
-  end note
```

Comment:
This (removed) note should remain because it might be useful when, in the future, FPC will support chaincode upgrades.
The reason is that upgrading an enclave requires to change the chaincode definition. But if we commit the new definition before the key distribution, we may lose access to the old enclave. That's why something should likely happen between approve and commit.

Reply:
Even with designated peer, to support upgrade we would need key-distribution which we say right now is not MVP? (but then non-upgradability (#313) is right now also MVP). 

In any case, your observation that upgrades could result in a atomicity issue which might require interleaving key-mgmt with commits is a good one. If we leave a corresponding comment (which seems a good idea), we should make it though a bit more explicit that we might need a kindof two-phase commit between the chaincode definition and ercc updates?

Reply:
will (hopefully :-) be addressed in upcoming separate PR ...